src/dataregistry/registrar/dataset.py

```diff
@@ -7,403 +7,481 @@
+
+		Only one dataset should ever be found.
+
+		Parameters
+		----------
+		relative_path : str, optional
+			Relative path to dataset
+		owner : str, optional
+			Owner of the dataset
+		owner_type : str, optional
+		dataset_id : int, optional
+
+		Returns
+		-------
+		r : CursorResult object
+			Searched dataset
+		"""
+
+		# Make sure we have all the relavant information
+		if dataset_id is None:
+			if (relative_path is None) or (owner is None) or (owner_type is None):
+				raise ValueError(
+					"Must pass relative_path, owner and owner_type to _find_entry"
+				)
+
+		# Search for dataset in the registry.
+		dataset_table = self._get_table_metadata("dataset")
+		stmt = select(dataset_table)
+
+		if dataset_id is None:
+			stmt = stmt.where(
+				dataset_table.c.relative_path == relative_path,
+				dataset_table.c.owner == owner,
+				dataset_table.c.owner_type == owner_type,
+			)
+		else:
+			stmt = stmt.where(dataset_table.c.dataset_id == dataset_id)
+
+		with self._engine.connect() as conn:
+			result = conn.execute(stmt)
+			conn.commit()
+
+		# Pull out the single result
+		for r in result:
+			return r
+
+		# No results found
+		return None
+
+	def delete(self, dataset_id):
+		"""
+		Delete an dataset entry from the DESC data registry.
+
+		This will also remove the raw data from the root dir, but the dataset
+		entry remains in the registry (now with an updated `status` field).
+
+		Parameters
+		----------
+		dataset_id : int
+			Dataset we want to delete from the registry
+		"""
+
+		# First make sure the given dataset id is in the registry
+		dataset_table = self._get_table_metadata(self.which_table)
+		previous_dataset = self._find_entry(dataset_table, dataset_id=dataset_id)
+
+		# Check dataset exists
+		if previous_dataset is None:
+			raise ValueError(f"Dataset ID {dataset_id} does not exist")
+		# Check dataset is valid
+		if not get_dataset_status(previous_dataset.status, "valid"):
+			raise ValueError(f"Dataset ID {dataset_id} does not have a valid status")
+		# Check dataset has not already been deleted
+		if get_dataset_status(previous_dataset.status, "deleted"):
+			raise ValueError(f"Dataset ID {dataset_id} does not have a valid status")
+
+		# Update the status of the dataset to deleted
+		with self._engine.connect() as conn:
+			update_stmt = (
+				update(dataset_table)
+				.where(dataset_table.c.dataset_id == dataset_id)
+				.values(
+					status=set_dataset_status(previous_dataset.status, deleted=True),
+					delete_date=datetime.now(),
+					delete_uid=self._uid,
+				)
+			)
+			conn.execute(update_stmt)
+			conn.commit()
+
+		# Delete the physical data in the root_dir
+		if previous_dataset.data_org != "dummy":
+			data_path = _form_dataset_path(
+				previous_dataset.owner_type,
+				previous_dataset.owner,
+				previous_dataset.relative_path,
+				schema=self._schema,
+				root_dir=self._root_dir,
+			)
+			print(f"Deleting data {data_path}")
+			os.remove(data_path)
```

Comment:
This only removes regular files.   If the path is the directory you'll need to do something else. `shutil.rmtree` looks like it does the right thing.

Reply:
Yes, well spotted.

I have changed to rmtree for directories, and added deleting a dir into the unit tests.